lib/ut/vec.c

```diff
@@ -113,6 +114,10 @@ void test_vec(void)
 	M0_UT_ASSERT(bv.ov_vec.v_nr == 0);
 	M0_UT_ASSERT(bv.ov_buf == NULL);
 	m0_bufvec_free(&bv);    /* no-op */
+	m0_fi_enable_once("bufvec_alloc", "buf-alloc-fail");
+	M0_UT_ASSERT(m0_bufvec_alloc(&bv, NR, M0_SEG_SIZE) == -ENOMEM);
+	m0_fi_enable_once("bufvec_alloc", "empty-fail");
+	M0_UT_ASSERT(m0_bufvec_empty_alloc(&bv, NR) == -ENOMEM);
```

Comment:
I know these two FI points are disabled automatically because they are triggered only once.
But do you think it's a good practice that we call m0_fi_disable() for them after use?

Reply:
Well, if they are disabled automatically - I don't see the point of calling `m0_fi_disable()` explicitly. Moreover, it might cause a confusion, since "enable_once" API implies that the fault injection will be enabled only once. And if it is not touched by the tested code (for example, when the execution did not pass that place where the fault injection is placed) - the UT will fail. Which is good, because it notifies developer that something is wrong with the test (or the code). With m0_fi_disable() there will be no such notification.

So I would say that the good practice is rather to not call m0_fi_disable() explicitly if m0_fi_enable_once() is used.